lib/src/objdetect/objdetect_async.dart

```diff
@@ -0,0 +1,808 @@
+library cv;
+
+import 'dart:ffi' as ffi;
+import 'dart:typed_data';
+import 'package:ffi/ffi.dart';
+import '../core/base.dart';
+import '../core/mat.dart';
+import '../core/point.dart';
+import '../core/rect.dart';
+import '../core/size.dart';
+import '../core/vec.dart';
+import '../opencv.g.dart' as cvg;
+
+class CascadeClassifier extends CvStruct<cvg.CascadeClassifier> {
+  CascadeClassifier._(cvg.CascadeClassifierPtr ptr, [bool attach = true])
+      : super.fromPointer(ptr) {
+    if (attach) {
+      finalizer.attach(this, ptr.cast(), detach: this);
+    }
+  }
+
+  factory CascadeClassifier.empty() {
+    final p = calloc<cvg.CascadeClassifier>();
+    cvRun(() => CFFI.CascadeClassifier_New(p));
+    return CascadeClassifier._(p);
+  }
+
+  factory CascadeClassifier.fromFile(String filename) {
+    final p = calloc<cvg.CascadeClassifier>();
+    final cp = filename.toNativeUtf8().cast<ffi.Char>();
+    cvRun(() => CFFI.CascadeClassifier_NewFromFile(cp, p));
+    calloc.free(cp);
+    return CascadeClassifier._(p);
+  }
+
+  Future<bool> loadAsync(String name) async {
+    final cname = name.toNativeUtf8().cast<ffi.Char>();
+    final rval = cvRunAsync<bool>(
+        (callback) => CFFI.CascadeClassifier_Load_Async(ref, cname, callback),
+        (c, p) {
+      final rval = p.cast<ffi.Int>().value != 0;
+      calloc.free(cname);
+      return c.complete(rval);
+    });
+    return rval;
+  }
+
+  Future<VecRect> detectMultiScaleAsync(InputArray image,
+      {double scaleFactor = 1.1,
+      int minNeighbors = 3,
+      int flags = 0,
+      (int, int) minSize = (0, 0),
+      (int, int) maxSize = (0, 0)}) async {
+    final ret = calloc<cvg.VecRect>();
+    final rval = cvRunAsync<VecRect>(
+        (callback) => CFFI.CascadeClassifier_DetectMultiScaleWithParams_Async(
+            ref,
+            image.ref,
+            scaleFactor,
+            minNeighbors,
+            flags,
+            minSize.cvd.ref,
+            maxSize.cvd.ref,
+            callback), (c, _) {
+      return c.complete(VecRect.fromPointer(ret));
+    });
+    return rval;
+  }
+
+  Future<(VecRect objects, VecInt numDetections)> detectMultiScale2Async(
+      InputArray image,
+      {double scaleFactor = 1.1,
+      int minNeighbors = 3,
+      int flags = 0,
+      (int, int) minSize = (0, 0),
+      (int, int) maxSize = (0, 0)}) async {
+    final ret = calloc<cvg.VecRect>();
+    final pnums = calloc<cvg.VecInt>();
+    final rval = cvRunAsync<(VecRect, VecInt)>(
+        (callback) => CFFI.CascadeClassifier_DetectMultiScale2_Async(
+            ref,
+            image.ref,
+            scaleFactor,
+            minNeighbors,
+            flags,
+            minSize.cvd.ref,
+            maxSize.cvd.ref,
+            ret,
+            pnums,
+            callback), (c, _) {
+      return c.complete((VecRect.fromPointer(ret), VecInt.fromPointer(pnums)));
+    });
+    return rval;
```

Comment:
Here, please always remember in the async version, no need to calloc pointers at dart side, the callbacks will return them from native side
```suggestion
    // if the callback returns nothing (i.e., CvCallback_0()), use cvRunAsync0, if returns 1 value (i.e., CvCallback_1(void *)), use cvRunAsync1, if returns 2 values (i.e., CvCallback_2(void *, void *)), use cvRunAsync2, etc.
    final rval = cvRunAsync2<(VecRect, VecInt)>(
        (callback) => CFFI.CascadeClassifier_DetectMultiScale2_Async(
            ref,
            image.ref,
            scaleFactor,
            minNeighbors,
            flags,
            minSize.cvd.ref,
            maxSize.cvd.ref,
            callback),
            (c, p1, p2) {
                // the returned pointer are all ffi.Pointer<ffi.Void>, so we need to cast them to the correct type
                return c.complete((VecRect.fromPointer(p1.cast<cvg.VecRect>()), VecInt.fromPointer(p2.cast<cvg.VecInt>())));
    });
    return rval;
```

Reply:
Oh got, thank you, will recheck everything hopefully all goes well

Reply:
yes, you are right